### ISSUE TITLE
[platform-server]: Proposal - ViewEncapsulation.Native errors in Node - no Html result

### ISSUE THREAD

**Issue by MarkPieszak** [angular/angular#14937]
**I'm submitting a ...** 
```
[x] bug report 
```

## Proposal
Maybe we could temporarily Render it as `ViewEncapsulation.Emulated` for only the Server-rendered view, and let only the Client use `Native`.

----

**Current behavior**
On a Component using `ViewEncapsulation.Native`, it errors on the server completely.
This is happening here in [`server_renderer`] since we're throwing an error.

(https://github.com/angular/angular/blob/126fda2613b1ad87ff62591bb1d3a23cd8f369a1/modules/%40angular/platform-server/src/server_renderer.ts#L44-L45) 

```
@Component({
    encapsulation: ViewEncapsulation.Native,
    ...
})
```
Error stacktrace:
```
EXCEPTION: Native encapsulation is not supported on the server!
ORIGINAL STACKTRACE:
Error: Native encapsulation is not supported on the server!
    at ServerRendererFactoryV2.createRenderer (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:36677:27) [angular]
    at createViewNodes (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:14118:70) [angular]
    at createRootView (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:14057:5) [angular]
    at Object.createProdRootView [as createRootView] (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:14613:12) [angular]
    at ComponentFactory_.create (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:11906:49) [angular]
    at ApplicationRef_.bootstrap (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:6554:60) [angular]
    at C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:6287:35 [angular]
    at Array.forEach (native) [angular]
    at PlatformRef_._moduleDoBootstrap (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:6286:46) [angular]
    at C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:6236:33 [angular]
    at Object.onInvoke (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:5396:41) [angular]
    at Zone.run (C:\Users\mark.pieszak\Documents\GitHub\ng-universal-demo\dist\server.js:51923:43) [angular => angular]
```

**Expected behavior**
To still be able to Render the result.

**Minimal reproduction of the problem with instructions**
https://github.com/MarkPieszak/ng-universal-demo/tree/native-encapsulation

**What is the motivation / use case for changing the behavior?**
Let people still be able to use Native, and simply use Emulated during the server render.

* **Angular version:** 4.0-rc2

* **Browser:** [all ]
 
cc/ @vikerman @alxhub @gdi2290

**Comment by gdi2290** [comment]
yeah in 2.1.x we switched ViewEncapsulation.Native to ViewEncapsulation.Emulated which can be done in the renderer

**Comment by IgorMinar** [comment]
closed by  #15155
